Source/src/resources/ResourceMaterial.h

```diff
@@ -91,20 +108,29 @@ namespace Hachiko
                 return "Specular";
             case ResourceTexture::Type::NORMALS:
                 return "Normals";
+            case ResourceTexture::Type::METALNESS:
+                return "Metalness";
             }
         }
 
         ResourceTexture* diffuse = nullptr;
         ResourceTexture* specular = nullptr;
         ResourceTexture* normal = nullptr;
+        ResourceTexture* metalness = nullptr;
         float4 diffuse_color = float4::one;
         float4 specular_color = float4::one / 10.0f;
-        float shininess = 50.0f;
+        float smoothness = 0.5f;
+        float metalness_value = 0.5f;
+        unsigned is_metallic = 0;
+        unsigned smoothness_alpha = 0;
 
     private:
         void AddTexture(ResourceTexture::Type type);
         void RemoveTexture(ResourceTexture::Type type);
         void UpdateMaterial();
         std::string name;
+
+        std::vector<char*> material_types = {"Specular", "Metallic"};
```

Comment:
would be better to use std::string instead of char*

Reply:
Done